### PR TITLE
DateInput picker-only mode

### LINKED ISSUE #1220: DateInput component should support picker only mode
We have a client that would like to not enable the direct editing of the text display, and force the user to pick with the picker.

Seemed like a very reasonable request that avoids users getting caught up in the wrong date formatting etc. etc.

### PR DESCRIPTION
**Pull request by febbraiod**
Hoist P/R Checklist
-------------------
Review and check off the below. Items that do not apply should be checked to indicate they have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Checked for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

Fixes #1220

### CODE REVIEW

**Comment by febbraiod** [comment]
There is a corresponding branch in toolbox for demo purpose, but not intended for merge

**Comment by lbwexler** [comment]
This looks good -- glad it was so straightforward.  Sorry I wasn't clearer about the name -- I think I was just trying to bang that ticket in quickly, and was not thinking that would be the name

**Comment by lbwexler** [comment]
Let's go for symmetry with the existing  'enablePicker' and call it 'enableTextInput'?
Obviously a bit weird if both were set false, but think that is going to be the most straightforward.

**Comment by lbwexler** [comment]
I went ahead and changed the name.

Testing out the formatting, I realized that it didn't look quite right.  It really should not be rendered as *disabled*.  That implies its fixed -- Its not!  It just should not be editable with the keyboard, thus encouraging/forcing the user to use the button.

**Comment by febbraiod** [comment]
10-4 will look at the styling vs disabled edit friday

**Comment by lbwexler** [comment]
Looking much better, but still not right.  Right now, if enableTextInput is false, you cannot tab into the control.  Realize this is extremely tricky and sheds light on the fact that our implementation might be seriously wrong for this case.  When enableTextInput is false, I think we just need an independent calendarButton  that is adjacent to a read only Text display, not an embedded icon.  

When they are both enabled, might use this new layout as well, or leave the current setup -- its really a matter of which will be easiest to implement and give a consistent look-and-feel.

File: desktop/cmp/input/DateInput.js [hunk 1]
```diff
@@ -172,22 +179,25 @@ export class DateInput extends HoistInput {
...
                     tabIndex: -1, // Prevent focus on tab
                     onClick: this.onClearBtnClick
                 }),
                 button({
-                    omit: !enablePicker,
+                    className: enablePicker ? '' : 'disabled-picker-icon',
                     icon: Icon.calendar(),
-                    tabIndex: -1, // Prevent focus on tab
-                    onClick: this.onPopoverBtnClick
+                    tabIndex: isPickerOnlyMode ? undefined : -1, // Prevent focus on tab, unless in pickerOnly mode
```
Comment (febbraiod) [inline desktop/cmp/input/DateInput.js]: This fixes our focus on tab problem without having to move the picker button out of the disabled input. A little embarrassed at how long this took me to figure out given the incredibly obvious comment that I apparently just didn't read about 100 times I looked at this code. :/

File: desktop/cmp/input/DateInput.js [hunk 2]
```diff
@@ -165,29 +173,34 @@ export class DateInput extends HoistInput {
                 })
             }),
 
+            onClick: !enableTextInput ? this.onOpenPopoverClick : null,
```
Comment (febbraiod) [inline desktop/cmp/input/DateInput.js]: Think this needs the isPickerOnlyProp.

**Pull request merged**